File: deploy/new-host.pl
#!/usr/bin/env perl
use 5.016;
use warnings;

use IPC::System::Simple;
use autodie qw(:all);

sub xqx {
    my $command = join ' ', @_;
    my $result = qx($command);
    if ($?) {
        die "Command '$command' failed";
    }
    return $result;
}

sub aws_credentials {
    my $content = xqx("cat $ENV{HOME}/.aws/credentials");
    my ($access_key) = $content =~ /aws_access_key_id = (.+)$/m or die "Can't find an access key";
    my ($access_secret) = $content =~ /aws_secret_access_key = (.+)$/m or die "Can't find a secret access key";
    return {
        key => $access_key,
        secret => $access_secret,
    };
}

sub create_ec2_host {
    my $aws = aws_credentials();
    my $VPC = 'vpc-488dc52d'; # berekuk's VPC
    system(
        'docker-machine', 'create',
        '--driver', 'amazonec2',
        '--amazonec2-access-key', $aws->{key},
        '--amazonec2-secret-key', $aws->{secret},
        '--amazonec2-vpc-id', $VPC,
        'questhub',
    );
}

sub main {
    create_ec2_host();
}

main unless caller;

File: deploy/new-host.py
#!/usr/bin/env python3

import re
import subprocess

def load_aws_credentials():
    content = subprocess.check_output('cat ~/.aws/credentials', shell=True, universal_newlines=True)
    access_key = re.search('aws_access_key_id = (.+)$', content, flags=re.MULTILINE).group(1)
    access_secret = re.search('aws_secret_access_key = (.+)$', content, flags=re.MULTILINE).group(1)
    return {
        'key': access_key,
        'secret': access_secret,
    }

def main():
    aws = load_aws_credentials()
    VPC = 'vpc-488dc52d' # berekuk's VPC
    subprocess.check_call([
        'docker-machine', 'create',
        '--driver', 'amazonec2',
        '--amazonec2-access-key', aws['key'],
        '--amazonec2-secret-key', aws['secret'],
        '--amazonec2-vpc-id', VPC,
        'questhub',
    ])


if __name__ == '__main__':
    main()
